sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/CheckAnalysis.scala

```diff
@@ -126,26 +126,11 @@ trait CheckAnalysis extends PredicateHelper with LookupCatalog {
         throw new IllegalStateException(
           "[BUG] logical plan should not have output of char/varchar type: " + leaf)
 
-      case u: UnresolvedNamespace =>
-        u.failAnalysis(s"Namespace not found: ${u.multipartIdentifier.quoted}")
-
-      case u: UnresolvedTable =>
-        u.failAnalysis(s"Table not found: ${u.multipartIdentifier.quoted}")
-
-      case u: UnresolvedView =>
-        u.failAnalysis(s"View not found: ${u.multipartIdentifier.quoted}")
-
-      case u: UnresolvedTableOrView =>
-        val viewStr = if (u.allowTempView) "view" else "permanent view"
-        u.failAnalysis(
-          s"Table or $viewStr not found: ${u.multipartIdentifier.quoted}")
```

Comment:
Why removing these?

Reply:
I am thinking this is because now we throw the exceptions from catalog implementation thus such unresolved namespace/table/view/function will be caught already in the earlier stage. So literally this PR moves to a fail-fast approach.


This makes sense to me as the giant `CheckAnalysis` sometime cannot give the error message I expect as there is an ordering of error check here. `Fail-fast` and `Fail-at-right-place` can improve it.

Reply:
Yea, not we fail fast and don't need the code here.